internal/ethapi/api_test.go

```diff
@@ -834,6 +839,50 @@ func TestEstimateGas(t *testing.T) {
 			},
 			want: 21000,
 		},
+		// Should be able to estimate SetCodeTx.
+		{
+			blockNumber: rpc.LatestBlockNumber,
+			call: TransactionArgs{
+				From:              &accounts[0].addr,
+				To:                &accounts[1].addr,
+				Value:             (*hexutil.Big)(big.NewInt(0)),
+				AuthorizationList: []types.SetCodeAuthorization{setCodeAuthorization},
+			},
+			want: 46000,
+		},
+		// Should retrieve the code of 0xef0001 || accounts[0].addr and return an invalid opcode error.
+		{
+			blockNumber: rpc.LatestBlockNumber,
+			call: TransactionArgs{
```

Comment:
This is a mind-twisting test but I think I get it now :)

Authorization: account0 to itself
Execute: recipient is account0, it will fetch the delegate which is again itself (recursive). So it will return the raw delegation code (prefix+account0) and try to execute it which will fail.

Is that the intention?

Reply:
Yes.